common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ShuffleIndexInformation.java

```diff
@@ -31,17 +34,44 @@
 public class ShuffleIndexInformation {
   /** offsets as long buffer */
   private final LongBuffer offsets;
+  private final boolean hasDigest;
+  /** digests as long buffer */
+  private final LongBuffer digests;
   private int size;
 
   public ShuffleIndexInformation(File indexFile) throws IOException {
+    ByteBuffer offsetsBuffer, digestsBuffer;
     size = (int)indexFile.length();
-    ByteBuffer buffer = ByteBuffer.allocate(size);
-    offsets = buffer.asLongBuffer();
+    int offsetsSize, digestsSize;
+    if (size % 8 == 0) {
```

Comment:
Would you please explain more about why using `size % 8` to check whether the index file has digest or not?

Reply:
has save digest to a independent file